src/DotNetty.Transport/Channels/DefaultChannelId.cs

```diff
@@ -0,0 +1,419 @@
+        static readonly Regex MachineIdPattern = new Regex("^(?:[0-9a-fA-F][:-]?){6,8}$");
+        const int MachineIdLen = 8;
+        static readonly byte[] MachineId;
+        const int ProcessIdLen = 4;
+        // Maximal value for 64bit systems is 2^22.  See man 5 proc.
+        // See https://github.com/netty/netty/issues/2706
+        const int MaxProcessId = 4194304;
+        static readonly int ProcessId;
+        const int SequenceLen = 4;
+        const int TimestampLen = 8;
+        const int RandomLen = 4;
+
+        static int nextSequence = 0;
+
+        static int seed = (int)(Stopwatch.GetTimestamp() & 0xFFFFFFFF); //used to safly cast long to int, because the timestamp returned is long and it doesn't fit into an int
+        static readonly ThreadLocal<Random> ThreadLocalRandom = new ThreadLocal<Random>(() => new Random(Interlocked.Increment(ref seed))); //used to simulate java ThreadLocalRandom
+
+        readonly byte[] data = new byte[MachineIdLen + ProcessIdLen + SequenceLen + TimestampLen + RandomLen];
+        int hashCode;
+
+        [NonSerialized]
+        string shortValue;
+
+        [NonSerialized]
+        string longValue;
+
+        static DefaultChannelId()
+        {
+            int processId = -1;
+            string customProcessId = SystemPropertyUtil.Get("io.netty.processId");
+            if (customProcessId != null)
+            {
+                int.TryParse(customProcessId, out processId);
+            }
+            if (processId < 0 || processId > MaxProcessId)
+            {
+                processId = -1;
+            }
+            if (processId < 0)
+            {
+                processId = DefaultProcessId();
+            }
+            ProcessId = processId;
+            byte[] machineId = null;
+            string customMachineId = SystemPropertyUtil.Get("io.netty.machineId");
+            if (customMachineId != null)
+            {
+                if (MachineIdPattern.Match(customMachineId).Success)
+                {
+                    machineId = ParseMachineId(customMachineId);
+                }
+            }
+
+            if (machineId == null)
+            {
+                machineId = DefaultMachineId();
+            }
+            MachineId = machineId;
+        }
+
+        static byte[] ParseMachineId(string value)
+        {
+            // Strip separators.
+            value = value.Replace("[:-]", "");
+            byte[] machineId = new byte[MachineIdLen];
+            for (int i = 0; i < value.Length; i += 2)
+            {
+                machineId[i] = (byte)int.Parse(value.Substring(i, i + 2), NumberStyles.AllowHexSpecifier);
+            }
+            return machineId;
+        }
+
+        static int DefaultProcessId()
+        {
+            int pId = -1;
+            try
+            {
+                pId = Process.GetCurrentProcess().Id;
+            }
+            catch
+            {
+                // ignored
```

Comment:
ops...it can be safely removed

On Mon, Nov 2, 2015 at 10:47 PM, Max Gortman <notifications@github.com>
wrote:

> In src/DotNetty.Transport/Channels/DefaultChannelId.cs
> <https://github.com/Azure/DotNetty/pull/11#discussion_r43687376>:
>
> > +            {
> > +                machineId[i] = (byte)int.Parse(value.Substring(i, i + 2), NumberStyles.AllowHexSpecifier);
> > +            }
> > +            return machineId;
> > +        }
> > +
> > +        static int DefaultProcessId()
> > +        {
> > +            int pId = -1;
> > +            try
> > +            {
> > +                pId = Process.GetCurrentProcess().Id;
> > +            }
> > +            catch
> > +            {
> > +                // ignored
>
> is try-catch needed here at all?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Azure/DotNetty/pull/11/files#r43687376>.
>